src/util/preferredLanguage.js

```diff
@@ -0,0 +1,15 @@
+import { hasExtension, getExtension } from "./extensions";
+
+export const getPreferredLanguage = hasExtension("getPreferredLanguage") ?
+  getExtension("getPreferredLanguage")
+  : () => {
+  let browserLang = window.navigator.language.split("-")[0],
```

Comment:
I think important to note that this only considers the most-preferred language, but it's critical to consider all the user's preferred languages ([`window.navigator.languages`](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/languages)) as their second or third preference might be the only one we support. Using a library like [accept-language](https://www.npmjs.com/package/accept-language) can help with this matching.